src/enzyme_ad/jax/Passes/ArithRaising.cpp

```diff
@@ -43,34 +44,76 @@ struct ArithRaisingPass
     auto op = getOperation();
 
-    op->walk([=](arith::AddFOp addOp) {
-      if (!addOp.getType().isa<RankedTensorType>())
-        return;
-      OpBuilder builder(addOp);
-      Value newAddOp;
-      if (use_stablehlo)
-        newAddOp = builder.create<stablehlo::AddOp>(
-            addOp.getLoc(), addOp->getOperand(0), addOp->getOperand(1));
-      else
-        newAddOp = builder.create<mhlo::AddOp>(
-            addOp.getLoc(), addOp->getOperand(0), addOp->getOperand(1));
-      addOp.replaceAllUsesWith(newAddOp);
-      addOp.erase();
-    });
-    op->walk([=](complex::AddOp addOp) {
-      if (!addOp->getResultTypes()[0].isa<RankedTensorType>())
-        return;
-      OpBuilder builder(addOp);
-      Value newAddOp;
-      if (use_stablehlo)
-        newAddOp = builder.create<stablehlo::AddOp>(
-            addOp.getLoc(), addOp->getOperand(0), addOp->getOperand(1));
-      else
-        newAddOp = builder.create<mhlo::AddOp>(
-            addOp.getLoc(), addOp->getOperand(0), addOp->getOperand(1));
-      addOp.replaceAllUsesWith(newAddOp);
-      addOp.erase();
-    });
+#define RAISE_BINARY(BinaryOp, StableHLOOp, MHLOOp)                            \
+  op->walk([=](BinaryOp origOp) {                                              \
+    if (!origOp->getResult(0).getType().isa<RankedTensorType>())               \
+      return;                                                                  \
+    OpBuilder builder(origOp);                                                 \
+    Value newOp;                                                               \
+    if (use_stablehlo)                                                         \
+      newOp = builder                                                          \
+                  .create<StableHLOOp>(origOp.getLoc(), origOp->getOperand(0), \
+                                       origOp->getOperand(1))                  \
+                  .getResult();                                                \
+    else                                                                       \
+      newOp = builder                                                          \
+                  .create<MHLOOp>(origOp.getLoc(), origOp->getOperand(0),      \
+                                  origOp->getOperand(1))                       \
+                  .getResult();                                                \
+    origOp.replaceAllUsesWith(newOp);                                          \
+    origOp.erase();                                                            \
+  });
+
+    RAISE_BINARY(arith::AddFOp, stablehlo::AddOp, mhlo::AddOp);
+    RAISE_BINARY(arith::AddIOp, stablehlo::AddOp, mhlo::AddOp);
+    RAISE_BINARY(arith::SubFOp, stablehlo::SubtractOp, mhlo::SubtractOp);
+    RAISE_BINARY(arith::SubIOp, stablehlo::SubtractOp, mhlo::SubtractOp);
+    RAISE_BINARY(arith::MulFOp, stablehlo::MulOp, mhlo::MulOp);
+    RAISE_BINARY(arith::MulIOp, stablehlo::MulOp, mhlo::MulOp);
```

Comment:
this is fine for now but I don't think we need to consider mhlo here long term, just stablehlo is fine